### ISSUE TITLE
Last Ordered Items block - bad js code 

### ISSUE THREAD

**Issue by Eytham** [magento/magento2#13157]
The left block Last Ordered Items is not showing properly the Add to cart button on the : 
data-bind="css: {'no-display': !lastOrderedItems().isShowAddToCart}
Always returning false.
Target JS file is : Magento_Sales/web/js/view/last-ordered-items.js

### Preconditions
1. Environnement Docker NGINX php 7
2. Magento 2.2

### Steps to reproduce
1. In my account or in catalog, left block Last ordered items not showing properly for add to cart button

### Expected result
1. Add to cart button viewable to add last ordered items to cart

### Actual result
1. lastOrderedItems().isShowAddToCart always returning FALSE so the add to cart div actions stands with no-display CSS class

### Original Code of Magento_Sales/web/js/view/last-ordered-items.js 
``` javascript
define([
    'uiComponent',
    'Magento_Customer/js/customer-data'
], function (Component, customerData) {
    'use strict';

    return Component.extend({
        /** @inheritdoc */
        initialize: function () {
            var isShowAddToCart = false,
                item;

            this._super();
            this.lastOrderedItems = customerData.get('last-ordered-items');
            
            for (item in this.lastOrderedItems.items) {
                if (item['is_saleable']) {
                    isShowAddToCart = true;
                    break;
                }
            }

            this.lastOrderedItems.isShowAddToCart = isShowAddToCart;
        }
    });
});
```

### My correction Code : 

``` javascript
define([
    'uiComponent',
    'Magento_Customer/js/customer-data',
    'jquery'
], function (Component, customerData, $) {
    'use strict';

    return Component.extend({
        /** @inheritdoc */
        initialize: function () {
            var isShowAddToCart = false,
                item;
            
            this._super();
            this.lastOrderedItems = customerData.get('last-ordered-items');
            
            var self = this;
            $(this.lastOrderedItems().items).each(function(){
                if(this['is_saleable']) {
                    self.lastOrderedItems().isShowAddToCart = true;
                }
            });
        }
    });
});
```

**Comment by ihor-sviziev** [comment]
@Eytham could you please create Pull request with your changes? It will be much easier to review your changes because we will see diff, all type of tests will be executed

**Comment by magento-engcom-team** [comment]
@Eytham, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by sidolov** [comment]
Hi @Eytham. Thank you for your report. 
The issue has been fixed in magento/magento2#19039 by @omiroshnichenko in 2.3-develop branch
Related commit(s):
 - [c333367742e668ea824c09675eb059fb246181a2](https://github.com/magento/magento2/commit/c333367742e668ea824c09675eb059fb246181a2)
 - [4803537b41f6ca2b059c2a64a3c2a03d7c9482e1](https://github.com/magento/magento2/commit/4803537b41f6ca2b059c2a64a3c2a03d7c9482e1)
 - [7c9de8e88705824b07378de2bca5ae383c5c80fb](https://github.com/magento/magento2/commit/7c9de8e88705824b07378de2bca5ae383c5c80fb)
 - [eaaed4b93ea1c716721d1b36ff9e6a9ea912c43c](https://github.com/magento/magento2/commit/eaaed4b93ea1c716721d1b36ff9e6a9ea912c43c)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @Eytham. Thank you for your report. 
The issue has been fixed in magento/magento2#19357 by @gelanivishal in 2.2-develop branch
Related commit(s):
 - [7559e3e1e3005b37750f07f791f8efaa0a0a6264](https://github.com/magento/magento2/commit/7559e3e1e3005b37750f07f791f8efaa0a0a6264)
 - [58550b9a406536cf8b6db8af2ca62d9f8a18aec9](https://github.com/magento/magento2/commit/58550b9a406536cf8b6db8af2ca62d9f8a18aec9)
 - [f1bb98ec0b2ffe12cd91b161f83bfd76cda0827f](https://github.com/magento/magento2/commit/f1bb98ec0b2ffe12cd91b161f83bfd76cda0827f)

The fix will be available with the upcoming 2.2.8 release.
